### PR TITLE
[runtime] Fix leaks exhibited by aot runtime corlib tests

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

**Comment by akoeplinger** [comment]
Unassigning myself as I'm not familiar with the runtime/AOT code 😄

File: mono/utils/mono-threads.c [hunk 1]
```diff
@@ -592,6 +592,7 @@ mono_thread_info_attach (void *baseptr)
 	info = (MonoThreadInfo *) mono_native_tls_get_value (thread_info_key);
 	if (!info) {
 		info = (MonoThreadInfo *) g_malloc0 (thread_info_size);
+		mono_native_tls_set_value (thread_info_key, info);
 		THREADS_DEBUG ("attaching %p\n", info);
```
Comment (vargaz) [inline mono/utils/mono-threads.c]: Why is this needed ? register_thread () already does this.

**Comment by vargaz** [comment]
Committed most of the changes.
